MultiplayerTestRunner/src/CLIArgs.cpp

```diff
@@ -0,0 +1,107 @@
+#include "CLIArgs.h"
+
+#include "../include/ErrorCodes.h"
+#include "../thirdparty/CLI11.hpp"
+
+#include <iostream>
+
+namespace
+{
+constexpr char* DEFAULT_TEST_ENDPOINT	 = "https://ogs-internal.magnopus-dev.cloud";
+constexpr int DEFAULT_TIMEOUT_IN_SECONDS = 30;
+
+/*
+ * Take the raw input from the CLI, validate, and convert to structured data.
+ * If Login or Password not provided, this will attempt to read `test_account_creds.txt`, and throw if it cannot.
+ * TestIdentifier is the only mandatory arg, this will validate that the string provided is an actual identifier.
+ * For optional arguments, if empty data is provided, will populate them with defaults.
+ * SpaceId will not be defaulted, and will be left empty if empty, in the expectation a space is about to be created and the value set externally.
+ */
+CLIArgs::RunnerSettings ValidateInvocationArgs(std::optional<std::string> LoginEmail,
+											   std::optional<std::string> LoginPassword,
+											   std::string TestIdentifier,
```

Comment:
[question] is it odd to have the only non-optional parameter in third place rather than first?